tests/www/apply/test_process.py

```diff
@@ -667,37 +667,222 @@ def test_process(self, *args, **kwargs):
         job_application = JobApplication.objects.get(pk=job_application.pk)
         assert job_application.state.is_processing
 
-    def test_refuse(self, *args, **kwargs):
-        """Ensure that the `refuse` transition is triggered."""
+    def test_refuse_session_prefix(self, *args, **kwargs):
+        """Ensure that each refusal session is isolated from each other."""
 
-        states = [
-            job_applications_enums.JobApplicationState.NEW,
-            job_applications_enums.JobApplicationState.PROCESSING,
-            job_applications_enums.JobApplicationState.PRIOR_TO_HIRE,
-            job_applications_enums.JobApplicationState.POSTPONED,
-        ]
+        job_application = JobApplicationFactory()
+        employer = job_application.to_company.members.first()
+        self.client.force_login(employer)
+        assert f"wizard_job_application_{job_application.pk}_refuse" not in self.client.session
 
-        for state in states:
-            with self.subTest(state=state):
-                job_application = JobApplicationFactory(sent_by_authorized_prescriber_organisation=True, state=state)
+        url = reverse("apply:refuse", kwargs={"job_application_id": job_application.pk, "step": "reason"})
+        self.client.get(url)
 
-                employer = job_application.to_company.members.first()
-                self.client.force_login(employer)
+        assert f"wizard_job_application_{job_application.pk}_refuse" in self.client.session
+        assert self.client.session[f"wizard_job_application_{job_application.pk}_refuse"] == {
+            "step": None,
+            "step_data": {},
+            "step_files": {},
+            "extra_data": {},
+        }
 
-                url = reverse("apply:refuse", kwargs={"job_application_id": job_application.pk})
-                response = self.client.get(url)
-                assert response.status_code == 200
+        post_data = {
+            f"job_application_{job_application.pk}_refuse-current_step": "reason",
+            "reason-refusal_reason": job_applications_enums.RefusalReason.HIRED_ELSEWHERE,
+        }
+        self.client.post(url, data=post_data)
+        assert f"wizard_job_application_{job_application.pk}_refuse" in self.client.session
+        assert self.client.session[f"wizard_job_application_{job_application.pk}_refuse"] == {
```

Comment:
Il y aura un `KeyError` si la clé n’est pas dans la session.
```suggestion
        assert self.client.session[f"wizard_job_application_{job_application.pk}_refuse"] == {
```